src/Tribe/Admin_List.php

```diff
@@ -175,22 +176,22 @@ public static function filter_by_aggregator_record( array $clauses, WP_Query $wp
 			}
 
 			// Check if filtering by aggregator record.
-			$record_num = (int) tribe_get_request_var( 'aggregator_record', 0 );
-			if ( 0 === $record_num ) {
+			$parent_record_id = (int) tribe_get_request_var( 'aggregator_record', 0 );
+			if ( 0 === $parent_record_id ) {
 				return $clauses;
 			}
 
 			global $wpdb;
-			
+
 			$table_alias = 'ea_record_' . substr( uniqid( 'ea_record', true ), 0, 10 );
 			// Add the record meta query if it is missing.
 			if ( ! preg_match( '/\\s' . preg_quote( $table_alias, '/' ) . '\\s/', $clauses['join'] ) ) {
-				$clauses['join'] .= " LEFT JOIN {$wpdb->postmeta} AS {$table_alias} ON {$wpdb->posts}.ID = {$table_alias}.post_id AND {$table_alias}.meta_key = '_tribe_aggregator_parent_record' ";
+				$clauses['join'] .= " LEFT JOIN {$wpdb->postmeta} AS {$table_alias} ON {$wpdb->posts}.ID = {$table_alias}.post_id AND {$table_alias}.meta_key = '_tribe_aggregator_record' ";
 			}
 
 			// Add the record meta filter if it is missing.
 			if ( ! preg_match( '/\\s' . preg_quote( $table_alias , '/' ) . '\\s/', $clauses['where'] ) ) {
-				$clauses['where'] .= " AND {$table_alias}.meta_value = '{$record_num}' ";
+				$clauses['where'] .= " AND {$table_alias}.meta_value IN ( SELECT {$wpdb->posts}.ID FROM {$wpdb->posts} WHERE {$wpdb->posts}.post_parent = {$parent_record_id} ) ";
```

Comment:
I think this one also faces the same "issue" where a parent record can have let's say 100 records but in this particular case you are interested on a chunk let's say the first 10 or the first 20 depending on the page you are looking for, is that correct?

Reply:
I do understand the concern, but there's not a way to know how many child records to pull. Most aggregator records only pull events that are in the future, so if we were to only pull (for example) 1 record, then ALL events in the past wouldn't be pulled. There might be only one event that would show if only one event were imported for that record, but there could be thousands of past events from previous imports that need to be detected to display the correct pagination numbers. By using the temporary table, we're filtering out events that don't include a child record ID as the import record. We're not pulling the child records into memory like e were before.  The query itself will still be limited to a certain number of events and only display the number of events based on the pagination settings.

Reply:
Right, I agree this is a great approach, however we are pulling all the records where the parent is `_parent_record_id`  which I think is great, however we don't need to pull all of them we just need to pull the first 20 (if this is the first page and the number of results per page are 20) meaning we are missing a `LIMIT` and  `OFFSET` depending of the current page and results per page into the sub query so it only returns that subset of records instead of all the ones associated with the parent ID.

Let me know if that makes sense. 

Additionally I think it might be good if we use `wpdb->prepare` to make sure expected values are passed into those variables. 

Great job, thanks and let me know if you have additional questions.

Reply:
Actually, I'm not altering the built in LIMIT and OFFSET at all. I'm only altering the 'where' and 'join' clauses. Not the 'limits' clause. It's the one that includes the LIMIT and OFFSET.

When outputting the 'limits' clause, it reads:
'limits' => 'LIMIT 0, 10'

My page limit is set to 10.

I hope that clears up the confusion.

Reply:
Correct, however the subquery (inside of the where) does not have this limit in consideration.

```sql
SELECT {$wpdb->posts}.ID FROM {$wpdb->posts} WHERE {$wpdb->posts}.post_parent = {$parent_record_id} )
```

So if this subquery will return all the records when not required so in this case you only need the first 10 and so on so this one is the one that might be missing the `LIMIT`

Reply:
Correct. That is because we want to only include records that are a part of ANY of the child records. Not just the first 20 or so. We are not "pulling" rows here. Only querying against the IDs that have a certain post_parent value.

Reply:
Great makes sense then, wanted to make sure that was the case :)

Reply:
Looking at it from a different vantage point: If we were to use LIMIT and OFFSET in the sub-query, (assuming posts per page is set to 20) when they go to page 2 of events, our subquery would then look for records 21 - 40. If there are only 5 aggregator records, it would return no children IDs and therefore all the events would disappear.

Reply:
I did add $wpdb->prepare() for the subquery.